src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -25,24 +26,41 @@ public class DeleteCommand extends Command {
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Entry: %1$s";
 
     private final Index targetIndex;
+    final EntryType entryType;
 
-    public DeleteCommand(Index targetIndex) {
+    public DeleteCommand(Index targetIndex, EntryType entryType) {
         this.targetIndex = targetIndex;
+        this.entryType = entryType;
     }
 
-    // TODO: Might need to edit
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        //List<Entry> lastShownList = model.getFilteredEntryList();
-        List<Entry> lastShownList = model.getFilteredExpenditureList();
+        List<Entry> lastShownList;
+        Entry entryToDelete;
+        switch (entryType.getEntryType()) {
+            case EXPENDITURE:
+                lastShownList = model.getFilteredExpenditureList();
+                if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+                }
```

Comment:
Possible to abstract his out as a method since it applies to both cases, or the switch case can just handle the assignment of the last shown list